app/models/host/managed.rb

```diff
@@ -551,11 +556,33 @@ def jumpstart?
     operatingsystem.family == "Solaris" and architecture.name =~/Sparc/i rescue false
   end
 
+  def hostgroup_inherited_attributes
+    %w{puppet_proxy_id puppet_ca_proxy_id environment_id compute_profile_id}
+  end
+
+  def apply_inherited_attributes(attributes, initialized = true)
+    return nil unless attributes
+    new_hostgroup_id = attributes['hostgroup_id']
```

Comment:
When changing `hostgroup_name` from the API, this isn't detected and so you don't get inheritance - both should work.